### PR TITLE
limit /tx_search output

### PR DESCRIPTION
**Pull request by melekes**
Refs #909

* [ ] Updated all relevant documentation in docs
* [ ] Updated all code comments where relevant
* [ ] Wrote tests
* [ ] Updated CHANGELOG.md

### CODE REVIEW

**Comment by melekes** [comment]
> We should also probably do this for /unconfirmed_txs, even though we're bounding the mempool, because the bound will be high.

yes, but only `per_page` / `limit` because mempool is constantly changing. don't see how someone will be able to paginate the unconfirmed txs.

**Comment by ebuchman** [comment]
> Don't see how someone will be able to paginate the unconfirmed txs.

Right. So do we just make it not possible to see all the txs ?

**Comment by melekes** [comment]
> So do we just make it not possible to see all the txs ?

Do we have a choice? The opposite is to allow unlimited txs, which can result in 10000? (mempool limit) entries in a worst case? Someone can probably exploit this. What's the use case for this endpoint? To ensure the tx I've just sent made it to the mempool?

**Comment by ebuchman** [comment]
> What's the use case for this endpoint? To ensure the tx I've just sent made it to the mempool?

Yeh, or to get a diagnostic of whats in the mempool.

Maybe we just need to make sure we can search the mempool for txs by hash and move /unconfirmed_Txs to  /unsafe_unconfirmed_txs

**Comment by melekes** [comment]
mm, actually calling `/unconfirmed_tx?hash` will result in whole list traversal (expensive!). unless we have a `map[hash]*pointer to CElement`. but not sure it's worth creating a map just because of 1 endpoint, which did not prove itself yet

File: rpc/core/tx.go [hunk 1]
```diff
@@ -182,11 +187,15 @@ func TxSearch(query string, prove bool) ([]*ctypes.ResultTx, error) {
...
+	totalCount := len(results)
+	page = validatePage(page)
+	perPage = validatePerPage(perPage)
+	skipCount := (page - 1) * perPage
+
+	apiResults := make([]*ctypes.ResultTx, cmn.MinInt(perPage, totalCount-skipCount))
 	var proof types.TxProof
-	for i, r := range results {
+	for i := 0; i < len(apiResults); i++ {
+		r := results[skipCount+i]
```
Comment (ebuchman) [inline rpc/core/tx.go]: this could be out of bounds, since page is not upper bounded

**Pull request merged**